KIALI-2034 Missing create change on clusterrole.yaml

lucasponce opened the pull request:
We have added a new endpoint for create operations but the changes on clusterrole.yaml were missed.
This is blocker for the feature.

cc @jmazzitelli @gbaufake as these are another changes in the pipeline for Maistra and Istio upstream.

lucasponce commented:
Note, we are not splitting create/update/patch from UI perspective, canUpdate will give grant to all three verbs.

If we need additional granularity we can add it in next iterations.

jotak commented on business/istio_config.go:
```diff
@@ -471,5 +473,5 @@ func getUpdateDeletePermissions(k8s kubernetes.IstioClientInterface, namespace,
 		}
 	}
-	return canUpdate || canPatch, canDelete
+	return canCreate || canUpdate || canPatch, canDelete
```
it seems weird to "merge" create permission with update/patch boolean.
How is it used on the frontend? a "create" button is displayed only if we have "create" permission, or is it something different?

lucasponce replied:
Create/Update are being using in similar scenarios, so it makes sense merge them at the moment. IMO

jotak replied:
ok, but then maybe it should be ANDed rather than ORed, because if you have update rights but not create rights, you should not be able to reach the create screen.

jotak commented on business/istio_config.go:
```diff
@@ -458,6 +458,8 @@ func getUpdateDeletePermissions(k8s kubernetes.IstioClientInterface, namespace,
 			for _, ssar := range ssars {
 				if ssar.Spec.ResourceAttributes != nil {
 					switch ssar.Spec.ResourceAttributes.Verb {
+					case "create":
```
line 456 is missing the "create" argument to k8s.GetSelfSubjectAccessReview.

lucasponce replied:
ups, true, good catch !

jmazzitelli commented on deploy/kubernetes/clusterrole.yaml:
```diff
@@ -78,6 +80,7 @@ rules:
   resources:
   - policies
   verbs:
+  - create
```
I added these three new "create" perms to my istio branch that I will use when I submit our PR for upgrading istio helm charts when we release kiali v0.14. See https://github.com/istio/istio/compare/master...jmazzitelli:kiali-next-version-changes?expand=1

gbaufake commented:
This PR will require some updates on Clusterole comparision (Currently, I am comparing Kiali master to Istio master) 

- https://github.com/kiali/kiali/blob/master/tests/e2e/tests/test_kiali_istio_clusterrole_binding.py

rhqci commented:
Jenkins CI: kiali-core-pr-e2e-test [#461](https://paste.fedoraproject.org/paste/oC-AfVHz1ize3Vz6S1Cz6g)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1075]](https://paste.fedoraproject.org/paste/wmj5Wu8KKMsjKM81mSyJBQ)

rhqci commented:
Jenkins CI: kiali-core-pr-e2e-test [#465](https://paste.fedoraproject.org/paste/1lxHRHj3B~BCP7BOK4Xzrg)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1079]](https://paste.fedoraproject.org/paste/nXLdv1LIqoiGuHIYaDlbvg)

lucasponce commented:
@jotak done, I've reviewed the verbs and we query the "update" and "patch" but we only use patch for modifications.
I guess we can also clean the "update" but I don't remember if there was some side effect with the rbac.
I think I addressed your comment, let me know.
Thanks

rhqci commented:
Jenkins CI: kiali-core-pr-e2e-test [#466](https://paste.fedoraproject.org/paste/mdPEvCxeuGZRFdLAFFt~kQ)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1080]](https://paste.fedoraproject.org/paste/YqZXtDEWqM3EmiuZcLc52w)

rhqci commented:
Jenkins CI: kiali-core-pr-e2e-test [#469](https://paste.fedoraproject.org/paste/HQCDoMPDpkHBpfZyjkSclA)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1085]](https://paste.fedoraproject.org/paste/b1uyymGk6f9ve2CBsFf6xQ)

rhqci commented:
Jenkins CI: kiali-core-pr-e2e-test [#470](https://paste.fedoraproject.org/paste/ihCdtciqnZWSTTNAh24Zhw)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1086]](https://paste.fedoraproject.org/paste/F6BC0QEe3u0KMPrBHt6OuQ)

lucasponce commented:
@jotak Finally I splitted the canCreate / canUpdate. It was a minor change after all and perhaps it's more clear now.

lucasponce commented:
True, thanks @jotak
